appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroViewPager.kt

```diff
@@ -97,45 +96,68 @@ class AppIntroViewPager(context: Context, attrs: AttributeSet) : ViewPager(conte
         customScroller?.scrollDurationFactor = factor
     }
 
-    override fun performClick() = super.performClick()
+    override fun onInterceptTouchEvent(event: MotionEvent?): Boolean
+    {
+        if (!handleTouchEvent(event)) {
+            return false
+        }
+
+        // Calling super will allow the slider to "work" left and right.
+        return super.onInterceptTouchEvent(event)
+    }
 
     override fun onTouchEvent(event: MotionEvent?): Boolean {
+        if (!handleTouchEvent(event)) {
+            return false
+        }
+
+        // Calling super will allow the slider to "work" left and right.
+        return super.onTouchEvent(event)
+    }
+
+    private fun handleTouchEvent(event: MotionEvent?) : Boolean {
```

Comment:
I would recommend a comment specifying this function.

Reply:
Done